doc/connectivity/networking/api/lwm2m.rst

```diff
@@ -410,6 +410,93 @@ value of 1 is ok here).
 
 For a more detailed LwM2M client sample see: :ref:`lwm2m-client-sample`.
 
+LwM2M engine and application events
+***********************************
+
+The Zephyr LwM2M engine defines events that can be sent back to the application via callback
+functions. The engine state machine shows when the events are spawned. Events depicted in
+the diagram are listed in the table. The events are prefixed with `LWM2M_RD_CLIENT_EVENT_`.
+
+.. figure:: images/lwm2m_engine_state_machine.png
+    :alt: LwM2M engine state machine
+
+    State machine for the LwM2M engine
+
+.. list-table:: LwM2M RD Client events
+   :widths: auto
+   :header-rows: 1
+
+   * - Event ID
+     - Event Name
+     - Description
+     - Actions
+   * - 0
+     - NONE
+     - No event
+     - Do nothing
+   * - 1
+     - BOOTSTRAP_REG_FAILURE
+     - Bootstrap registration failed.
+       Happens if timeout or failure in bootstrap registration.
+     - Retry bootstrap
+   * - 2
+     - BOOTSTRAP_REG_COMPLETE
+     - Bootstrap registration complete.
+       Happens after successful bootstrap registration.
+     - No actions needed
+   * - 3
+     - BOOTSTRAP_TRANSFER_COMPLETE
+     - Bootstrap finish command received from the server.
```

Comment:
```suggestion
     - Bootstrap complete command received from the server.
```

Reply:
The operation is really named `Bootstrap-Finish`
https://www.openmobilealliance.org/release/LightweightM2M/V1_1_1-20190617-A/HTML-Version/OMA-TS-LightweightM2M_Core-V1_1_1-20190617-A.html#6-1-7-2-0-6172-Bootstrap-Finish-Operation